src/preferences/colorpaletteeditormodel.cpp

```diff
@@ -117,24 +132,78 @@ void ColorPaletteEditorModel::setColorPalette(const ColorPalette& palette) {
     setDirty(false);
 }
 
-ColorPalette ColorPaletteEditorModel::getColorPalette(const QString& name) const {
+ColorPalette ColorPaletteEditorModel::getColorPalette(
+        const QString& name) const {
     QList<mixxx::RgbColor> colors;
     QMap<int, int> hotcueColorIndices;
     for (int i = 0; i < rowCount(); i++) {
         QStandardItem* pColorItem = item(i, 0);
-        QStandardItem* pHotcueIndexItem = item(i, 1);
-        mixxx::RgbColor::optional_t color = mixxx::RgbColor::fromQString(pColorItem->text());
+        HotcueIndexListItem* pHotcueIndexItem =
+                static_cast<HotcueIndexListItem>(item(i, 1));
+        mixxx::RgbColor::optional_t color =
+                mixxx::RgbColor::fromQString(pColorItem->text());
         if (color) {
+            QList<int> hotcueIndexes =
+                    parseHotcueIndexList(pHotcueIndexItem->text());
             colors << *color;
 
-            bool ok;
-            int hotcueIndex = pHotcueIndexItem->text().toInt(&ok);
-            if (ok) {
-                hotcueColorIndices.insert(hotcueIndex - 1, colors.size() - 1);
+            for (int i = 0; i < hotcueIndexes.size(); ++i) {
+                hotcueColorIndices.insert(
+                        hotcueIndexes.at(i) - 1, colors.size() - 1);
             }
         }
     }
     // If we have a non consequitive list of hotcue indexes, indexes are shifted down
     // due to the sorting nature of QMap. This is intended, this way we have a color for every hotcue.
     return ColorPalette(name, colors, hotcueColorIndices.values());
 }
+
+HotcueIndexListItem::HotcueIndexListItem(const QList<int>& hotcueList = {})
+        : QStandardItem(), m_hotcueIndexList(hotcueList) {
+}
+QVariant HotcueIndexListItem::data(int role = Qt::UserRole + 1) const {
```

Comment:
My understanding here is that this acts as a getter for certain roles, right? Can't I just reuse the displayrole here and call the parent for every other role?

Reply:
https://github.com/mixxxdj/mixxx/blob/472ec12326c2fadc3441c1e7a31488e1877d43f3/src/library/browse/browsethread.cpp#L95
I think you want the display role like here.